client-secret-basic-poc/app.js

```diff
@@ -0,0 +1,80 @@
+const express = require('express');
+const crypto = require('crypto');
+const base64 = require('urlsafe-base64');
+const querystring = require('querystring');
+const axios = require('axios');
+
+const clientId = "8036ffddb8811020dbf3eb507d1949a9";
+const clientSecret = "8036ffddb8811020dbf3eb507d1949a9-8036ffddb8811020dbf3eb507d1949a9-8036ffddb8811020dbf3eb507d1949a9";
+const redirectUri = "https://idpartner-rp2.ngrok.io/button/oauth/callback";
+
+// Generate random values for state, nonce, and verifier
+const state = crypto.randomBytes(16).toString('hex');
+const nonce = crypto.randomBytes(16).toString('hex');
+const verifier = base64.encode(crypto.randomBytes(32));
+const challenge = base64.encode(crypto.createHash('sha256').update(verifier).digest());
+
+const app = express();
+
+// Endpoint to initiate the OAuth process
+app.get('/button/oauth', async (req, res) => {
+  // Construct query parameters for the authorization request
+  const queryParams = querystring.stringify({
+    redirect_uri: redirectUri,
+    code_challenge_method: "S256",
+    code_challenge: challenge,
+    state,
+    nonce,
+    scope: "openid offline_access email profile birthdate address",
+    client_id: clientId,
+    client_secret: clientSecret,
+    identity_provider_id: 1, // mikomo
```

Comment:
why is this needed?

Reply:
to complete the oauth flow without using the account selector